### PR TITLE
Fix make proof

### PR DESCRIPTION
**Pull request by dongrie**
Fix make proof

ibc-mock-client
https://github.com/datachainlab/ibc-mock-client/pull/11

### CODE REVIEW

File: tests/chains/tendermint/go.mod [hunk 1]
```diff
@@ -168,5 +168,6 @@ require (
 replace (
 	// https://github.com/cosmos/cosmos-sdk/blob/v0.47.3/go.mod#L171-L182
 	github.com/99designs/keyring => github.com/cosmos/keyring v1.2.0
+	github.com/datachainlab/ibc-mock-client => github.com/dongrie/ibc-mock-client v0.3.3-0.20231213015729-29b70e76baf2
```
Comment (dongrie) [inline tests/chains/tendermint/go.mod:171]: https://github.com/datachainlab/ibc-mock-client/pull/11
Once this PR is merged, we plan to switch to the new version.

File: provers/mock/prover.go [hunk 2]
```diff
@@ -117,10 +118,31 @@ func (pr *Prover) CheckRefreshRequired(dst core.ChainInfoICS02Querier) (bool, er
...
+func makeProof(height exported.Height, path string, bz []byte) []byte {
+	revisionNumber := height.GetRevisionNumber()
+	revisionHeight := height.GetRevisionHeight()
+
+	heightBig := new(big.Int)
+	revisionNumberBig := new(big.Int).SetUint64(revisionNumber)
+	revisionNumberBig = revisionNumberBig.Lsh(revisionNumberBig, 64)
+	revisionHeightBig := new(big.Int).SetUint64(revisionHeight)
+	heightBig.Or(revisionNumberBig, revisionHeightBig)
+	hashHeight := sha256.Sum256(heightBig.Bytes())
```
Comment (siburu) [inline provers/mock/prover.go:134]: The new spec is `sha256(height ++ sha256(prefix) ++ sha256(path) ++ sha256(value))`.
You should not hash the height here.

I think you can use `encoding/binary` to do this easilier.
https://pkg.go.dev/encoding/binary#example-ByteOrder-Put

Comment (dongrie) [reply provers/mock/prover.go]: Use encoding/binary
https://github.com/hyperledger-labs/yui-relayer/pull/129/commits/309fad3580e668febcf3c3b9a5f40124e7bc69c4

File: tests/chains/tendermint/go.mod [hunk 3]
```diff
@@ -168,5 +168,6 @@ require (
 replace (
 	// https://github.com/cosmos/cosmos-sdk/blob/v0.47.3/go.mod#L171-L182
 	github.com/99designs/keyring => github.com/cosmos/keyring v1.2.0
+	github.com/datachainlab/ibc-mock-client => github.com/dongrie/ibc-mock-client v0.3.3-0.20231221102347-9000a7df65b4
```
Comment (siburu) [inline tests/chains/tendermint/go.mod:171]: @dongrie I've just created the tag v0.3.3 of ibc-mock-client.
Please use ibc-mock-client@v0.3.3

File: yui-relayer
Comment (siburu) [inline yui-relayer:1]: Never commit this binary file.

**Comment by dongrie** [comment]
@siburu changed it to use ibc-mock-client@v0.3.3

**Review by siburu — APPROVED** [review]
@dongrie LGTM, thank you!

**Pull request merged**